lib/v2/parliament/section77.js

```diff
@@ -0,0 +1,191 @@
+            }[presenter] ?? '';
+
+        title +=
+            {
+                m: ' (ประธานสภาผู้แทนราษฎรวินิจฉัยว่า เป็นร่างการเงิน)',
+                nm: ' (ประธานสภาผู้แทนราษฎรวินิจฉัยว่า ไม่เป็นร่างการเงิน)',
+            }[isMonetaryAct] ?? '';
+    }
+
+    const result = {
+        title,
+        link: `${baseUrl}/survey_more_news.php${type ? '?type=' + type : ''}`,
+        language: 'th-th',
+        item: [],
+    };
+
+    // Fetch the news first so we can know how many pages do we need to fetch
+    const browser = await require('@/utils/puppeteer')();
+    const gotByPPT = async (url) => {
+        const page = await browser.newPage();
+        // intercept all requests
+        await page.setRequestInterception(true);
+        // only allow certain types of requests to proceed
+        page.on('request', (request) => {
+            // in this case, we only allow document requests to proceed
+            request.resourceType() === 'document' ? request.continue() : request.abort();
+        });
+        const response = await page.goto(url, {
+            // specify how long to wait for the page to load
+            waitUntil: 'domcontentloaded',
+        });
+        await page.setContent((await response.buffer()).toString('utf8'));
+
+        // retrieve the HTML content of the page
+        const html = await page.content();
+        // close the tab
+        page.close();
+
+        return html;
+    };
+
+    const response = await gotByPPT(result.link);
+    const $ = cheerio.load(response);
+    let pageCnt = 1;
+    const maxPageElem = $('#pagination-more_new a[id]:last-child');
+
+    if (maxPageElem.length) {
+        pageCnt = parseInt(maxPageElem.attr('id') ?? '1') || 1;
+    }
+
+    const actList = await Promise.all(
+        [...Array(pageCnt).keys()].map(async (pageIdx) => {
+            const queryParams = {
+                page: pageIdx + 1,
+                type,
+            };
+            if (type) {
+                queryParams.type = type;
+            }
+
+            const url = `${baseUrl}/ajax/pagination.php?${new URLSearchParams(queryParams).toString()}`;
+            const response = await gotByPPT(url);
+            const $ = cheerio.load(response);
+
+            if ($.text() === 'ยังไม่มีข้อมูล') {
+                return [];
+            }
+
+            return $('div.item-77')
+                .toArray()
+                .map((item) => {
+                    item = $(item);
+                    return {
+                        title: item.find('a').text(),
+                        link: `${baseUrl}/${item.find('a').attr('href')}`,
+                        category: item
+                            .find('label')
+                            .toArray()
+                            .map((l) => $(l).text()),
+                    };
+                });
+        })
+    ).then((actPageList) => actPageList.reduce((actPage, actPageAggr) => actPageAggr.concat(actPage), []));
```

Comment:
Please extract data from the first page only.

Reply:
Sorry for late reply.

Is it okay to leave the page without description?

Reply:
Oh, I see. Will remove additional pages then.